src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -0,0 +1,138 @@
+        Map<String, teammates.storage.sqlentity.Section> sectionNameToSectionMap = migrateSectionChain(newCourse);
+        System.out.println(sectionNameToSectionMap); // To stop lint from complaining
+        // TODO: Add mirgrateFeedbackChain
+        // migrateFeedbackChain(sectionNameToSectionMap);
+    }
+
+    private Map<String, teammates.storage.sqlentity.Section> migrateSectionChain(
+            teammates.storage.sqlentity.Course newCourse) {
+        List<CourseStudent> oldStudents = ofy().load().type(CourseStudent.class).filter("courseId", newCourse.getId())
+                .list();
+        Map<String, teammates.storage.sqlentity.Section> sections = new HashMap<>();
+        Map<String, List<CourseStudent>> sectionToStuMap = oldStudents.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getSectionName));
+
+        for (Map.Entry<String, List<CourseStudent>> entry : sectionToStuMap.entrySet()) {
+            String sectionName = entry.getKey();
+            List<CourseStudent> stuList = entry.getValue();
+            teammates.storage.sqlentity.Section newSection = createSection(newCourse, sectionName);
+            sections.put(sectionName, newSection);
+            migrateTeams(newCourse, newSection, stuList);
+            saveEntityDeferred(newSection);
+        }
+        return sections;
+    }
+
+    private void migrateTeams(teammates.storage.sqlentity.Course newCourse,
+            teammates.storage.sqlentity.Section newSection, List<CourseStudent> studentsInSection) {
+        Map<String, List<CourseStudent>> teamNameToStuMap = studentsInSection.stream()
+                .collect(Collectors.groupingBy(CourseStudent::getTeamName));
+        for (Map.Entry<String, List<CourseStudent>> entry : teamNameToStuMap.entrySet()) {
+            String teamName = entry.getKey();
+            List<CourseStudent> stuList = entry.getValue();
+            teammates.storage.sqlentity.Team newTeam = createTeam(newSection, teamName);
+            migrateStudent(newCourse, newTeam, stuList);
+            saveEntityDeferred(newTeam);
+        }
+    }
+
+    private void migrateStudent(teammates.storage.sqlentity.Course newCourse, teammates.storage.sqlentity.Team newTeam,
+            List<CourseStudent> studentsInTeam) {
+        for (CourseStudent oldStudent : studentsInTeam) {
+            teammates.storage.sqlentity.Student newStudent = createStudent(newCourse, newTeam, oldStudent);
+            saveEntityDeferred(newStudent);
+        }
+    }
+
+    private teammates.storage.sqlentity.Course createCourse(Course oldCourse) {
+        teammates.storage.sqlentity.Course newCourse = new teammates.storage.sqlentity.Course(
+                oldCourse.getUniqueId(),
+                oldCourse.getName(),
+                oldCourse.getTimeZone(),
+                oldCourse.getInstitute());
+        newCourse.setDeletedAt(oldCourse.getDeletedAt());
+        // newCourse.setCreatedAt(oldCourse.getCreatedAt());
```

Comment:
Could we include the timestamp changes into this PR too? 

We can just remove the `@CreationTimestamp` in `BaseEntity`, then start including `setCreatedAt` and verifying it too. Similar for `@UpdateTimestamp` and `setUpdatedAt`.

Reply:
Good catch!